kinder/hack/orderimports/orderimports.go

```diff
@@ -0,0 +1,463 @@
+	flag.PrintDefaults()
+}
+
+func filterFile(f fs.DirEntry) bool {
+	// ignore non-Go files
+	name := f.Name()
+	if f.IsDir() ||
+		strings.HasPrefix(name, ".") ||
+		!strings.HasSuffix(name, ".go") {
+		return true
+	}
+	return false
+}
+
+// If in == nil, the source is the contents of the file with the given filename.
+func processFile(filename string, in io.Reader, out io.Writer, stdin bool) error {
+	fileSet := token.NewFileSet() // per file FileSet
+
+	var perm fs.FileMode = 0644
+	if in == nil {
+		f, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		fi, err := f.Stat()
+		if err != nil {
+			return err
+		}
+		in = f
+		perm = fi.Mode().Perm()
+	}
+
+	src, err := io.ReadAll(in)
+	if err != nil {
+		return err
+	}
+
+	astFile, err := parser.ParseFile(fileSet, filename, src, parserMode)
+	if err != nil {
+		return err
+	}
+
+	sortImports(fileSet, astFile)
+
+	var buf bytes.Buffer
+	cfg := printer.Config{Mode: printerMode, Tabwidth: tabWidth}
+	err = cfg.Fprint(&buf, fileSet, astFile)
+	if err != nil {
+		return err
+	}
+	res := buf.Bytes()
+
+	if !bytes.Equal(src, res) {
+		// formatting has changed
+		if *list {
+			fmt.Fprintln(out, filename)
+		}
+		if *write {
+			// make a temporary backup before overwriting original
+			bakname, err := backupFile(filename+".", src, perm)
+			if err != nil {
+				return err
+			}
+			err = os.WriteFile(filename, res, perm)
+			if err != nil {
+				os.Rename(bakname, filename)
+				return err
+			}
+			err = os.Remove(bakname)
+			if err != nil {
+				return err
+			}
+		}
+		if *doDiff {
+			data, err := diffWithReplaceTempFile(src, res, filename)
+			if err != nil {
+				return fmt.Errorf("computing diff: %s", err)
+			}
+			fmt.Printf("diff -u %s %s\n", filepath.ToSlash(filename+".orig"), filepath.ToSlash(filename))
+			out.Write(data)
+		}
+	}
+	if !*list && !*write && !*doDiff {
+		_, err = out.Write(res)
+	}
+
+	return err
+}
+
+func sortImports(fset *token.FileSet, f *ast.File) {
+	for _, d := range f.Decls {
+		d, ok := d.(*ast.GenDecl)
+		if !ok || d.Tok != token.IMPORT {
+			// Not an import declaration, so we're done.
+			// Imports are always first.
+			break
+		}
+		if !d.Lparen.IsValid() {
+			// Not a block: sorted by default.
+			continue
+		}
+		if len(d.Specs) <= 1 {
+			continue
+		}
+		d.Specs = sortDecl(fset, f, d)
+	}
+}
+
+func sortDecl(fSet *token.FileSet, f *ast.File, d *ast.GenDecl) []ast.Spec {
+	// split all imports into different groups
+	var stdlibImports, localImports, k8sImports, externalImports []*ast.ImportSpec
+	for _, spec := range d.Specs {
+		imp := spec.(*ast.ImportSpec)
+		importPath := strings.Replace(imp.Path.Value, "\"", "", -1)
+		parts := strings.Split(importPath, "/")
+		if !strings.Contains(parts[0], ".") {
+			// standard library
+			stdlibImports = append(stdlibImports, imp)
+		} else if strings.HasPrefix(importPath, "k8s.io/kubeadm") {
+			// local imports
+			localImports = append(localImports, imp)
```

Comment:
i think we should make the tool support any local prefix, potentially via a new flag.
this means that i could build the tool and call:
`orderimports.go -p k8s.io/kubeadm ...`

which would order with `k8s.io/kubeadm` being the local import path prefix.
autodetecting this seems trickier, but perhaps you have ideas @xinydev ?

(not sure if `-p` is right package short flag  vs e.g. `-local`)

in theory this would allow us to use the tool for any repo under k8s.

Reply:
adding a new flag is a good idea!